correct NU3024 message for validation scenario

heng-liu opened the pull request:
Bug
Fixes: NuGet/Home/[#7933](https://github.com/NuGet/Home/issues/7933)
Regression: No
Fix
Correct NU3024 message for two scenarios.
In signing scenario, change the message string to make it more precise.
In validation scenario, add a new error message string, as it should not be the same with NU3022.
Testing/Validation
Tests Added: No

joelverhagen committed:
Add PackageDetailsUriTemplate/5.1.0 resource and plumb through URL to UI (#2518)

Address https://github.com/NuGet/Home/issues/5299
Address https://github.com/NuGet/Home/issues/7470

nkolev92 committed:
Initial implementation of the core logic for PackageDownload. API plumbing and assets file/package spec writing update (#2688)

nkolev92 committed:
Refactor the nomination code. Add more tests (#2696)

nkolev92 committed:
Add support for reading the PackageDownload items (#2694)

* the targets now read the package download items similar to the PackageReference items

heng-liu committed:
correct NU3024 message for validation scenario

nkolev92 commented on src/NuGet.Core/NuGet.Packaging/Signing/Utility/VerificationUtility.cs:
```diff
@@ -144,7 +144,7 @@ internal static SignatureVerificationStatusFlags ValidateTimestamp(Timestamp tim
                 if (!spec.AllowedHashAlgorithmOids.Contains(signerInfo.DigestAlgorithm.Value))
                 {
-                    issues.Add(SignatureLog.Issue(treatIssuesAsErrors, NuGetLogCode.NU3024, string.Format(CultureInfo.CurrentCulture, Strings.VerifyError_TimestampUnsupportedSignatureAlgorithm, signature.FriendlyName)));
+                    issues.Add(SignatureLog.Issue(treatIssuesAsErrors, NuGetLogCode.NU3024, string.Format(CultureInfo.CurrentCulture, Strings.VerifyError_TimestampUnsupportedDigestAlgorithm, signature.FriendlyName)));
```
You are passing the signature.FriendlyName here but there's no placeholder for it in the message.

heng-liu replied:
Thanks! I'll fix it.

dtivel commented on src/NuGet.Core/NuGet.Packaging/Strings.resx:
```diff
@@ -227,6 +227,9 @@
     <comment>0 - Signature friendly name</comment>
   </data>
+  <data name="VerifyError_TimestampUnsupportedDigestAlgorithm" xml:space="preserve">
+    <value>The timestamp signature has an unsupported digest algorithm.</value>
```
Follow the example of the above resource string (`VerifyError_TimestampUnsupportedSignatureAlgorithm`):

`The {0}'s timestamp signature has an unsupported digest algorithm.`

That way, the full error message will be one of the following:

* The author primary signature's timestamp signature has an unsupported digest algorithm.
* The repository primary signature's timestamp signature has an unsupported digest algorithm.
* The repository countersignature's timestamp signature has an unsupported digest algorithm.

heng-liu replied:
Thanks for your explanation! I'll fix it.

zivkan committed:
Increment version to 5.1.0 (#2742)

nkolev92 committed:
Update SPDX license list (#2733)

heng-liu committed:
add placeholder for signature.FriendlyName

jeffkl committed:
Ignore case when looking for package restored by SDK resolver (#2734)

* Ignore case when looking for package restored by SDK resolver

If a user specifies a different case for an SDK name, initial restore fails because our check is case sensitive.

Fixes #7817

* Refactor code to use LibraryIdentity class to get better casing and version comparisons

* Don't allow SDK package version to float

WonyoungChoi committed:
Make Tizen6 map to NetStandard 2.1 (#2716)

nkolev92 commented on src/NuGet.Core/NuGet.Packaging/Strings.resx:
```diff
@@ -227,6 +227,10 @@
     <comment>0 - Signature friendly name</comment>
   </data>
+  <data name="VerifyError_TimestampUnsupportedDigestAlgorithm" xml:space="preserve">
+    <value>The {0}'s timestamp signature has an unsupported digest algorithm.</value>
```
Should this be enhanced by calling out the name of the digest algorithm? 

@dtivel

dtivel replied:
I think the OID is sufficient.

dtivel replied:
There are lots of places where we could do this, so I'm OK with making this change here and everywhere in a separate PR.  But open a separate issue for it.

nkolev92 replied:
Created an issue for that https://github.com/NuGet/Home/issues/7926

campersau committed:
implement V2 search metadata versions (#2569)

* don't throw NRE in PackageSearchMetadataV2Feed.GetVersionsAsync

* implement GetVersionsAsync for PackageSearchMetadataV2Feed by using V2FeedUtilities.CreatePackageSearchResult

* add public constructor for PackageSearchMetadataV2Feed back

nkolev92 committed:
Add new VS nomination APIs that contain changes for PackageDownload (#2705)

nkolev92 committed:
Fix the failing PackageMetadataResource tests - tests should not expect a type that's not part of the method signature (#2748)

nkolev92 committed:
Add a dirty check for the dg spec json

nkolev92 committed:
add tests

nkolev92 committed:
Update ToolPathResolverTests.cs

nkolev92 committed:
fx the tests

nkolev92 committed:
Merge branch 'dev-nkolev92-dirtyCheckForDgSpecJson' of https://github.com/NuGet/NuGet.Client into dev-nkolev92-dirtyCheckForDgSpecJson

dominoFire committed:
Adds non-zero exit code when the number of arguments are not appropiate (#2505)

Adds non-zero exit code when the number of arguments are not appropriate (#2505) (NuGet/Home#7178)

nkolev92 committed:
Add a dirty check for the dgspec.json write (#2750)

* Add a dirty check for the dg spec json

* add tests

* Update ToolPathResolverTests.cs

* fx the tests

dtivel committed:
Plugins:  enable diagnostic logging (#2753)

Resolve https://github.com/NuGet/Home/issues/7859.

dtivel committed:
Plugins:  improve logging (#2755)

Resolve https://github.com/NuGet/Home/issues/7859.

dtivel committed:
Test:  attempt to fix a flaky test (#2757)

Resolve https://github.com/NuGet/Home/issues/7877.

jeffkl committed:
Fix issue casing issue with GeneratePathProperty (#2746)

* Fix issue casing issue with GeneratePathProperty

Use a case insenstive HashSet instead of an IEnumerable<string>.  This removes the need to change the case comparison and should improve performance.

Also updated unit test

* Use different collections for packages with tools and packages with GeneratePathProperty set to true

dtivel committed:
Test:  fix flaky signing test (#2759)

Resolve https://github.com/NuGet/Home/issues/7879.

dtivel committed:
Test: add symbols to EndToEnd.zip (#2761)

Resolve https://github.com/NuGet/Home/issues/7885.

dtivel committed:
Test:  update E2E test for VS 2019 (#2762)

Resolve https://github.com/NuGet/Home/issues/7886.

dtivel committed:
Test:  improve test performance testing script (#2760)

nkolev92 committed:
Add a plugin deprecation message for the old plugin model (#2763)

nkolev92 committed:
NoOp for package download (#2767)

dtivel committed:
Plugins:  improve plugin logging datetime accuracy (#2771)

Resolve https://github.com/NuGet/Home/issues/7899.

nkolev92 committed:
Conditionally import the pack targets in the solution build (#2770)

nkolev92 committed:
fix the comparison to allow the generated nuspec to be read by the manifest reader (#2769)

dtivel committed:
Plugin:  log more plugin instance events (#2773)

Resolve https://github.com/NuGet/Home/issues/7907.

dtivel committed:
Settings:  avoid crash and improve error message (#2768)

Resolve https://github.com/NuGet/Home/issues/7820.

nkolev92 committed:
Use Directory.CreateDirectory instead of DirectoryUtility to create the obj folder  (#2774)

nkolev92 committed:
Improve the error case handling in the ManifestMetadata (#2776)

zivkan committed:
Use project full path as document path when opening PM UI for project (#2777)

You can right click the window tab in Visual Studio and select "Copy
Full Path" or "Open Containing Folder". VS needs to know the full path
to the project file for this to work.

nkolev92 committed:
Fix lucene signing (#2780)

nkolev92 committed:
FrameworkReference support in pack & restore (#2745)

nkolev92 committed:
Merge pull request #2766 from NuGet/dev-feature-PackageDownload

Merging features PackageDownload & FrameworkReference

nkolev92 reviewed (approved):
👏

zivkan committed:
Explain why we don't use the full path to execute `sh` (#2782)

heng-liu committed:
change remarks,error msg, string resource name for NU3024

nkolev92 committed:
Deprecate project.json pack (#2783)

dtivel commented on src/NuGet.Core/NuGet.Common/Errors/NuGetLogCode.cs:
```diff
@@ -361,7 +361,7 @@ public enum NuGetLogCode
 
         /// <summary>
-        /// The timestamp signing certificate has an unsupported signature algorithm.
+        /// The timestamp signature has an unsupported signature algorithm.
```
The timestamp signature has an unsupported digest algorithm.

dtivel reviewed (approved):
I think your change will break [this test](https://github.com/NuGet/NuGet.Client/blob/8d6b7ff631d103b2a2005ccd8b98aca642ccff8e/test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/SignCommandTests.cs#L710).  Make sure your test results are clean before merging.

dtivel committed:
Test:  improve E2E failure diagnostics (#2781)

Resolve https://github.com/NuGet/Home/issues/7925.

heng-liu committed:
correct remarks

heng-liu committed:
Correct error message NU3024

heng-liu committed:
Merge branch 'dev-hengliu-checkBug' of https://github.com/NuGet/NuGet.Client into dev-hengliu-checkBug

heng-liu committed:
fix 2 tests by changing exception messages

heng-liu commented:
Created another PR instead: https://github.com/NuGet/NuGet.Client/pull/2786
